code/datums/storage/storage.dm

```diff
@@ -0,0 +1,1027 @@
+	 * A reference to the atom linked to this storage object
+	 * If the parent goes, we go. Will never be null.
+	 */
+	VAR_FINAL/atom/parent
+	/**
+	 * A reference to the atom where the items are actually stored.
+	 * By default this is parent. Should generally never be null.
+	 * Sometimes it's not the parent, that's what is called "dissassociated storage".
+	 *
+	 * Do NOT set this directly, use set_real_location.
+	 */
+	VAR_FINAL/atom/real_location
+
+	/// Typecache of items that can be inserted into this storage.
+	/// By default, all item types can be inserted (assuming other conditions are met).
+	/// Do not set directly, use set_holdable
+	VAR_FINAL/list/obj/item/can_hold
+	/// Typecache of items that cannot be inserted into this storage.
+	/// By default, no item types are barred from insertion.
+	/// Do not set directly, use set_holdable
+	VAR_FINAL/list/obj/item/cant_hold
+	/// Typecache of items that can always be inserted into this storage, regardless of size.
+	VAR_FINAL/list/obj/item/storage_type_limits
+
+	/**
+	 * Associated list of types and their max count, formatted as
+	 * 	storage_type_limits_max = list(
+	 * 		/obj/A = 3,
+	 * 	)
+	 *
+	 * Any inserted objects will decrement the allowed count of every listed type which matches or is a parent of that object.
+	 * With entries for both /obj/A and /obj/A/B, inserting a B requires non-zero allowed count remaining for, and reduces, both.
+	 */
+	VAR_FINAL/list/storage_type_limits_max
+	///Max size of objects that this object can store (in effect only if canhold isn't set)
+	var/max_w_class = WEIGHT_CLASS_SMALL
+	///The sum of the storage costs of all the items in this storage item.
+	var/max_storage_space = 14
+	///The number of storage slots in this container.
+	var/storage_slots = 7
+	///Defines how many versions of the sprites that gets progressively emptier as they get closer to "_0" in .dmi.
+	var/sprite_slots = null
+
+	///In slotless storage, stores areas where clicking will refer to the associated item
+	var/list/click_border_start = list()
+	///In slotless storage, stores areas where clicking will refer to the associated item
+	var/list/click_border_end = list()
+	///storage UI
+	var/atom/movable/screen/storage/boxes = null
+	///storage UI
+	var/atom/movable/screen/storage/storage_start = null
+	///storage UI
+	var/atom/movable/screen/storage/storage_continue = null
+	///storage UI
+	var/atom/movable/screen/storage/storage_end = null
+	///storage UI
+	var/atom/movable/screen/storage/stored_start = null
+	///storage UI
+	var/atom/movable/screen/storage/stored_continue = null
+	///storage UI
+	var/atom/movable/screen/storage/stored_end = null
+	///storage UI
+	var/atom/movable/screen/close/closer = null
+
+	///whether our storage box on hud changes color when full.
+	var/show_storage_fullness = TRUE
+	///Set this to make it possible to use this item in an inverse way, so you can have the item in your hand and click items on the floor to pick them up.
+	var/use_to_pickup
```

Comment:
there is a lack of consistency in your init here that is weird.

Is there a way to have flags rather than bools ?

I will not force this, if you think it is too tedious I will live with it